src/source/PeerConnection/PeerConnection.c

```diff
@@ -1351,3 +1383,50 @@ STATUS deinitKvsWebRtc(VOID)
     LEAVES();
     return retStatus;
 }
+
+STATUS twccManagerOnPacketSent(PKvsPeerConnection pc, PRtpPacket pRtpPacket)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    BOOL locked = FALSE;
+    UINT64 sn = 0;
+    UINT16 seqNum;
+    BOOL isEmpty = FALSE;
+    INT64 firstTimeKvs, lastLocalTimeKvs, ageOfOldest;
+    CHK(pc != NULL && pRtpPacket != NULL, STATUS_NULL_ARG);
+    CHK(TWCC_EXT_PROFILE == pRtpPacket->header.extensionProfile, STATUS_SUCCESS);
+
+    MUTEX_LOCK(pc->twccLock);
+    locked = TRUE;
+
+    seqNum = TWCC_SEQNUM(pRtpPacket->header.extensionPayload);
+    CHK_STATUS(stackQueueEnqueue(&pc->twccManager.twccPackets, seqNum));
+    pc->twccManager.twccPacketBySeqNum[seqNum].seqNum = seqNum;
+    pc->twccManager.twccPacketBySeqNum[seqNum].packetSize = pRtpPacket->payloadLength;
+    pc->twccManager.twccPacketBySeqNum[seqNum].localTimeKvs = pRtpPacket->sentTime;
+    pc->twccManager.twccPacketBySeqNum[seqNum].remoteTimeKvs = TWCC_PACKET_LOST_TIME;
+    pc->twccManager.lastLocalTimeKvs = pRtpPacket->sentTime;
+
+    // cleanup queue until it contains up to 2 seconds of sent packets
+    do {
+        CHK_STATUS(stackQueuePeek(&pc->twccManager.twccPackets, &sn));
+        firstTimeKvs = pc->twccManager.twccPacketBySeqNum[(UINT16) sn].localTimeKvs;
+        lastLocalTimeKvs = pRtpPacket->sentTime;
+        ageOfOldest = lastLocalTimeKvs - firstTimeKvs;
+        if (ageOfOldest > TWCC_ESTIMATOR_TIME_WINDOW) {
+            CHK_STATUS(stackQueueDequeue(&pc->twccManager.twccPackets, &sn));
```

Comment:
`twccManager.twccPackets` must be cleared and freed using `stackQueueClear()` and `stackQueueFree()` at application cleanup in `freePeerConnection()` and/or when its usage is done.

Reply:
twccPackets is a `StackQueue` not `PStackQueue`. You think I need to `stackQueueFree` it as well?